Framework/Algorithms/src/CalculateIqt.cpp

```diff
@@ -0,0 +1,305 @@
+  IAlgorithm_sptr FFTAlgorithm =
+      AlgorithmManager::Instance().create("ExtractFFTSpectrum");
+  FFTAlgorithm->setChild(true);
+  FFTAlgorithm->initialize();
+  FFTAlgorithm->setProperty("InputWorkspace", workspace);
+  FFTAlgorithm->setProperty("OutputWorkspace", "_");
+  FFTAlgorithm->setProperty("FFTPart", 2);
+  FFTAlgorithm->execute();
+  return FFTAlgorithm->getProperty("OutputWorkspace");
+}
+
+MatrixWorkspace_sptr divide(MatrixWorkspace_sptr lhsWorkspace,
+                            MatrixWorkspace_sptr rhsWorkspace) {
+  IAlgorithm_sptr divideAlgorithm =
+      AlgorithmManager::Instance().create("Divide");
+  divideAlgorithm->setChild(true);
+  divideAlgorithm->initialize();
+  divideAlgorithm->setProperty("LHSWorkspace", lhsWorkspace);
+  divideAlgorithm->setProperty("RHSWorkspace", rhsWorkspace);
+  divideAlgorithm->setProperty("OutputWorkspace", "_");
+  divideAlgorithm->execute();
+  return divideAlgorithm->getProperty("OutputWorkspace");
+}
+
+MatrixWorkspace_sptr cropWorkspace(MatrixWorkspace_sptr workspace,
+                                   double xMax) {
+  IAlgorithm_sptr cropAlgorithm =
+      AlgorithmManager::Instance().create("CropWorkspace");
+  cropAlgorithm->setChild(true);
+  cropAlgorithm->initialize();
+  cropAlgorithm->setProperty("InputWorkspace", workspace);
+  cropAlgorithm->setProperty("OutputWorkspace", "_");
+  cropAlgorithm->setProperty("XMax", xMax);
+  cropAlgorithm->execute();
+  return cropAlgorithm->getProperty("OutputWorkspace");
+}
+
+MatrixWorkspace_sptr replaceSpecialValues(MatrixWorkspace_sptr workspace) {
+  IAlgorithm_sptr specialValuesAlgorithm =
+      AlgorithmManager::Instance().create("ReplaceSpecialValues");
+  specialValuesAlgorithm->setChild(true);
+  specialValuesAlgorithm->initialize();
+  specialValuesAlgorithm->setProperty("InputWorkspace", workspace);
+  specialValuesAlgorithm->setProperty("OutputWorkspace", "_");
+  specialValuesAlgorithm->setProperty("InfinityValue", 0.0);
+  specialValuesAlgorithm->setProperty("BigNumberThreshold", 1.0001);
+  specialValuesAlgorithm->setProperty("NaNValue", 0.0);
+  specialValuesAlgorithm->execute();
+  return specialValuesAlgorithm->getProperty("OutputWorkspace");
+}
+
+std::string createRebinString(double minimum, double maximum, double width) {
+  std::stringstream rebinStream;
+  rebinStream.precision(14);
+  rebinStream << minimum << ", " << width << ", " << maximum;
+  return rebinStream.str();
+}
+
+template <typename Generator>
+void randomizeHistogramWithinError(HistogramY &row, const HistogramE &errors,
+                                   Generator &generator) {
+  for (auto i = 0u; i < row.size(); ++i) {
+    auto randomValue = generator(errors[i]);
+    row[i] += randomValue;
+  }
+}
+
+MatrixWorkspace_sptr
+randomizeWorkspaceWithinError(MatrixWorkspace_sptr workspace, const int seed) {
+  MersenneTwister mTwister(seed);
+  auto rng = [&mTwister](const double error) {
```

Comment:
Does this stand for random number generator? I only realised that after googling Mersenne Twister, it looks like its supposed to mean range.

Reply:
Good point, it does look suspiciously like a range. I'll change it to the full name.